### PR TITLE
Fix pull request check list is limited

### LINKED ISSUE #25990: Pull request check list is limited to 30 entries
### Description

This is linked to the DEFAULT_PAGING_NUM value.
The computation of the check summary status is also wrong due to this limit: A status can be in progress and the summary value having the value "All checks were successful"

### Gitea Version

1.19

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Docker

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by CaiCandong**
In the original implementation, we can only get the first 30 records of the commit status (the default paging size), if the commit status is more than 30, it will lead to the bug #25990. I made the following two changes.
- On the page, use the ` db.ListOptions{ListAll: true}` parameter instead of `db.ListOptions{}`
-  The `GetLatestCommitStatus` function makes a determination as to whether or not a pager is being used.  

fixed #25990

### CODE REVIEW

File: models/git/commit_status.go [hunk 1]
```diff
@@ -283,9 +283,9 @@ func GetLatestCommitStatus(ctx context.Context, repoID int64, sha string, listOp
 		Where("repo_id = ?", repoID).And("sha = ?", sha).
 		Select("max( id ) as id").
 		GroupBy("context_hash").OrderBy("max( id ) desc")
-
-	sess = db.SetSessionPagination(sess, &listOptions)
-
+	if !listOptions.IsListAll() {
+		sess = db.SetSessionPagination(sess, &listOptions)
```
Comment (lunny) [inline models/git/commit_status.go:287]: How about put the detection into the function `SetSessionPagination`?

Comment (CaiCandong) [reply models/git/commit_status.go:287]: I was going to do the same at first, but I referenced how it was written elsewhere.
https://github.com/go-gitea/gitea/blob/73fb1ecdcf7c72c9b15caa70a8e3f444d311bd7b/models/git/branch_list.go#L106-L115

Comment (CaiCandong) [reply models/git/commit_status.go:287]: To avoid triggering other bugs, I've made minimal changes

**Comment by CaiCandong** [comment]
The current way of calculating combined status is not elegant enough. Perhaps we could have another implementation, but this would require more changes.

https://github.com/go-gitea/gitea/issues/25990#issuecomment-1653023772

**Review by lunny — APPROVED** [review]

**Review by denyskon — APPROVED** [review]

**Pull request merged**